explore/src/main/scala/explore/config/AdvancedConfigurationPanel.scala

```diff
@@ -181,11 +281,42 @@ sealed abstract class AdvancedConfigurationPanelBuilder[
           if (newExpTime === expTimeOverride.get) Callback.empty
           else expTimeOverride.set(newExpTime)
       )
+      // filter the spectroscopy matrix by the requirements that don't get overridden
+      // by the advanced config (wavelength, for example).
+      .useMemoBy((props, _, _) =>
+        (props.spectroscopyRequirements.focalPlane,
+         props.spectroscopyRequirements.capabilities,
+         props.spectroscopyRequirements.focalPlaneAngle,
+         props.spectroscopyRequirements.resolution,
+         props.spectroscopyRequirements.wavelengthCoverage
+        )
+      ) { (props, _, _) =>
+        { case (fp, cap, fpa, res, cov) =>
+          props.ctx.staticData.spectroscopyMatrix.filtered(
+            focalPlane = fp,
+            capabilities = cap,
+            slitWidth = fpa,
+            resolution = res,
+            coverage = cov.map(_.micrometer.toValue[BigDecimal].toRefined[Positive])
+          )
+        }
+      }
+      // Try to find the readonly data from the spectroscopy matrix
+      .useMemoBy { (props, _, _, _) =>
+        // TODO: reuse on rows, not requirements? And narrow advanced comparison?
+        (props.scienceModeBasic, props.scienceModeAdvanced.get, props.spectroscopyRequirements)
```

Comment:
If this looks OK to Andy, I should probably not base the reuse here on all of scienceModeAdvanced, but only the fields we use for the table lookup. 
I'm not sure if it it is worth basing the reuse on the `rows` from the previous `useMemoBy`. We'd need to add `Eq` instances to a whole bunch of classes in `SpectroscopyModesMatrix.scala`, and then it seems like we might as well just do the search rather than checking to see if the lists are equal. Thoughts?

Reply:
I'm not sure I understand what you are suggesting, but `useMemo` already returns a `Reusable` (which will be based in its own dependencies).

Reply:
Right. But, instead of checking the filtered list or spectroscopy rows for reusability now, I'm just checking the spectroscopy requirements upon which is is based. Currently, there is no Eq for any of the many types in SpectroscopyModesMatrix, and I'm not sure it's worth scanning the filtered table for equality (reusability) each time.

Reply:
Oh, I see what you are saying. I don't need a reuse of SpectroscopyModeRow, because the Reuse returned by `useMemo` is based on the dependecies of that `useMemo`, not the value of the memo. Nice.

Reply:
Pushed a commit to utilize that.